src/katib.py

```diff
@@ -0,0 +1,160 @@
+import json
+from functools import wraps
+from textwrap import dedent
+from typing import List
+
+from .data import Data
+
+
+def trial_template(name, image, replicas, command, mount_path, pr=None):
+    target = "pull/{pr}/head:{pr}".format(pr=pr) if pr else "master"
+    revision = pr if pr else "master"
+    args = ["cd data/kubernetes-analysis ;",
+            "git fetch origin {} ;".format(target),
+            "git checkout {} ;".format(revision),
+            "./main tune {{- with .HyperParameters}} {{- range .}} {{.Name}}={{.Value}} {{- end}} {{- end}}"]
+
+    return {
+        "apiVersion": "kubeflow.org/v1",
+        "kind": "TFJob",
+        "metadata": {
+            "name": "{{.Trial}}",
+            "namespace": "{{.NameSpace}}"
+        },
+        "spec": {
+            "tfReplicaSpecs": {
+                "Worker": {
+                    "replicas": replicas,
+                    "restartPolicy": "OnFailure",
+                    "template": {
+                        "spec": {
+                            "containers": [{
+                                "name": name,
+                                "image": image,
+                                "imagePullPolicy": "Always",
+                                "command": ["bash", "-c"],
+                                "args": args,
+                                "volumeMounts": [{
+                                    "mountPath": "/data",
+                                    "name": "nfs-storage"
+                                }]
+                            }],
+                            "volumes": [{
+                                "name": "nfs-storage",
+                                "persistentVolumeClaim": {
+                                    "claimName": "pipeline-pv"
+                                }
+                            }]
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+class KatibOp():
+    is_created = False
+
+    def __init__(self, image: str, name, output: str, repo: str, pr: str):
+        self.repo = repo
+        self.pr = pr
+        self.output = output
+        self.name = name
+        self.image = image
+        self.controller_img = "mbu93/katib-launcher:latest"
```

Comment:
Please commit the container file (Dockerfile) into the PR as well as the source code of the ml_runner.py.